--- a/drivers/net/wireless/intel/iwlwifi/iwl-nvm-parse.c
+++ b/drivers/net/wireless/intel/iwlwifi/iwl-nvm-parse.c
@@ -911,14 +911,9 @@
 {
 	bool is_ap = iftype_data->types_mask & (BIT(NL80211_IFTYPE_AP) |
 						BIT(NL80211_IFTYPE_P2P_GO));
-	bool no_320;
 
-	no_320 = (!trans->trans_cfg->integrated &&
-		 trans->pcie_link_speed < PCI_EXP_LNKSTA_CLS_8_0GB) ||
-		 trans->reduced_cap_sku;
-
-	if (!data->sku_cap_11be_enable || iwlwifi_mod_params.disable_11be)
-		iftype_data->eht_cap.has_eht = false;
+	/* Force no_320 to false to enable 320 MHz channels */
+	bool no_320 = false;
 
 	/* Advertise an A-MPDU exponent extension based on
 	 * operating band
